pom.xml

```diff
@@ -461,6 +463,32 @@
       </build>
     </profile>
 
+    <!-- Slightly adjust build to support building with JDK >= 21
+      However, by default this will intentionally be blocked by the Maven Enforcer Plugin defined above
+      and must be explicitly circumvented to avoid building non-release Gson artifacts by accident -->
+    <profile>
+      <id>jdk21+</id>
+      <activation>
+        <jdk>[21,)</jdk>
+      </activation>
+      <properties>
+        <!-- JDK 21 does not support Java 7 as release, must use at least Java 8 -->
+        <maven.compiler.release>8</maven.compiler.release>
+      </properties>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-compiler-plugin</artifactId>
+            <configuration>
+              <!-- Don't fail on warning because javac reports lint warnings for generated code of 'proto' module -->
```

Comment:
Can we fix those warnings? Not necessarily now, but perhaps an issue could track this.

Reply:
Ah nevermind, this is actually fixed in newer Protobuf versions: https://github.com/protocolbuffers/protobuf/issues/9673

The problem is that the Protobuf version we are using is outdated. Thanks for your question, which prompted me to properly investigate this!